src/Core/Feature.js

```diff
@@ -0,0 +1,154 @@
+        this.extent = feature.extent ? defaultExtent(feature.crs) : undefined;
+        this.indices = [];
+        this._feature = feature;
+        this.properties = {};
+    }
+
+    startSubGeometry(count) {
+        const last = this.indices.length - 1;
+        const offset = last > -1 ?
+            this.indices[last].offset + this.indices[last].count :
+            this._feature.vertices.length / this._feature.size;
+        this.indices.push({ offset, count });
+        _extendBuffer(this._feature, count);
+    }
+
+    closeSubGeometry(count) {
+        const last = this.indices.length - 1;
+        const offset = last > -1 ?
+            this.indices[last].offset + this.indices[last].count :
+            this._feature.vertices.length / this._feature.size - count;
+        this.indices.push({ offset, count });
+    }
+
+    pushCoordinates(coord) {
+        if (coord.crs !== this._feature.crs) {
+            coord.as(this._feature.crs, coord);
+        }
+        if (this._feature.normals) {
+            coord.geodesicNormal.toArray(this._feature.normals, this._feature._pos);
+        }
+
+        this._feature._pushValues(this._feature, coord._values[0], coord._values[1], coord._values[2]);
+        // expand extent if present
+        if (this.extent) {
+            this.extent.expandByCoordinates(coord);
+        }
+    }
+
+    pushCoordinatesValues(value0, value1, value2) {
+        this._feature._pushValues(this._feature, value0, value1, value2);
+        // expand extent if present
+        if (this.extent) {
+            this.extent.expandByValuesCoordinates(value0, value1, value2);
+        }
+    }
+
+    get size() {
+        return this._feature.size;
+    }
+}
+
+function push2DValues(feature, value0, value1) {
+    feature.vertices[feature._pos++] = value0;
+    feature.vertices[feature._pos++] = value1;
+}
+
+function push3DValues(feature, value0, value1, value2) {
+    feature.vertices[feature._pos++] = value0;
+    feature.vertices[feature._pos++] = value1;
+    feature.vertices[feature._pos++] = value2;
+}
+
+export default class Feature {
+    constructor(type, crs, options) {
+        this.type = type;
+        this.geometry = [];
+        this.vertices = [];
+        this.vertices.crs = crs;
+        this.normals = options.withNormal ? [] : undefined;
+        this.crs = crs;
+        this.size = options.withAltitude ? 3 : 2;
+        this.extent = options.buildExtent ? defaultExtent(crs) : undefined;
+        this._pos = 0;
+        this._pushValues = this.size === 3 ? push3DValues : push2DValues;
+    }
+
+    bindNewGeometry() {
+        const geometry = new FeatureGeometry(this);
+        this.geometry.push(geometry);
+        return geometry;
+    }
+
+    updateExtent(geometry) {
+        if (this.extent) {
+            unionExtent(this.extent, geometry.extent);
+        }
+    }
+
+    get geometryCount() {
+        return this.geometry.length;
+    }
+}
+
+export class MultiFeatures {
```

Comment:
What about using `FeatureCollection` like it is now ?

Reply:
yes